Orange/clustering/clustering.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+from Orange.data import Table, Instance
+from Orange.preprocess import Continuize, SklImpute
+
+
+def preprocess_data(preprocessors, data):
+    if preprocessors is not None:
+        for pp in preprocessors:
+            data = pp(data)
+    return data
+
+
+class ClusteringModel:
+
+    def __init__(self, projector, pre_domain, preprocessors):
+        self.projector = projector
+        self.pre_domain = pre_domain
+        self.preprocessors = preprocessors
+        self.labels = projector.labels_
+
+    def __call__(self, data):
+        if isinstance(data, Table):
+            if data.domain is not self.pre_domain:
+                data = data.transform(self.pre_domain)
+            data = preprocess_data(self.preprocessors, data)
+            return self.predict(data.X)
+        elif isinstance(data, Instance):
+            if data.domain is not self.pre_domain:
+                data = Instance(self.pre_domain, data)
+            return self.predict(data.x)
+        elif isinstance(data, np.ndarray):
+            return self.predict(data)
+        else:
+            raise Exception("The type of your data is not supported.")
+
+    def predict(self, X):
+        raise NotImplementedError(
+            "This clustering algorithm does not support predicting.")
+
+
+class Clustering:
+    """
+    ${skldoc}
```

Comment:
This remains in the docstrings of subclasses. Check e.g. `help(Orange.clustering.DBSCAN)`